cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,302 @@
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.cloudbreak.cloud.model.filesystem.CloudFileSystemView;
+import com.sequenceiq.cloudbreak.cloud.model.filesystem.CloudS3View;
+import com.sequenceiq.cloudbreak.cloud.model.SpiFileSystem;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+import com.sequenceiq.cloudbreak.validation.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.common.api.cloudstorage.AccountMappingBase;
+import com.sequenceiq.common.api.cloudstorage.StorageLocationBase;
+import com.sequenceiq.common.model.CloudIdentityType;
+import com.sequenceiq.common.model.CloudStorageCdpService;
+
+public class AwsIDBrokerObjectStorageValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsIDBrokerObjectStorageValidator.class);
+
+    // The data access and ranger audit lists need to match the IDBMMS lists
+    // https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/
+    //  services/idbrokermappingmanagement/src/main/java/com/cloudera/thunderhead/
+    //  service/idbrokermappingmanagement/server/IdBrokerMappingManagementService.java#L109
+    private static final List<String> DATA_ACCESS_USERS = Arrays.asList("hbase", "hdfs", "hive",
+        "impala", "yarn", "dpprofiler", "zeppelin", "kudu", "hue");
+
+    private static final List<String> RANGER_AUDIT_USERS = Arrays.asList("kafka", "solr", "knox",
+        "atlas");
+
+    private final AmazonIdentityManagement iam;
+
+    public AwsIDBrokerObjectStorageValidator(AmazonIdentityManagement amazonIdentityManagement) {
+        this.iam = amazonIdentityManagement;
+    }
+
+    public ValidationResult validateObjectStorage(SpiFileSystem spiFileSystem, ValidationResultBuilder resultBuilder) {
+        List<CloudFileSystemView> cloudFileSystems = spiFileSystem.getCloudFileSystems();
+        for (CloudFileSystemView cloudFileSystemView : cloudFileSystems) {
+            CloudS3View cloudFileSystem = (CloudS3View) cloudFileSystemView;
+            InstanceProfile instanceProfile = getInstanceProfile(cloudFileSystem.getInstanceProfile(),
+                resultBuilder);
+            if (instanceProfile != null) {
+                CloudIdentityType cloudIdentityType = cloudFileSystem.getCloudIdentityType();
+                if (CloudIdentityType.ID_BROKER.equals(cloudIdentityType)) {
+                    checkInstanceProfileTrust(instanceProfile, resultBuilder);
+                    checkIDBrokerAssumeRolePermissions(instanceProfile, cloudFileSystem, resultBuilder);
+                    checkDataAccessRolePermissions(spiFileSystem, cloudFileSystem, resultBuilder);
+                    checkRangerAuditRolePermissions(spiFileSystem, cloudFileSystem, resultBuilder);
+                } else if (CloudIdentityType.LOG.equals(cloudIdentityType)) {
+                    checkInstanceProfileTrust(instanceProfile, resultBuilder);
```

Comment:
Though not related to IDBroker, it might be beneficial to check log bucket permissions as well as per `aws-cdp-log-policy.json`.

Reply:
yea that was supposed to happen (hence why I copied the json file). I'll add that too. Looks like I missed it when going through the first time.

Reply:
So this is harder than it looks. Not sure where the log location is stored. I put a placeholder in with a TODO to find how to pass in the log location.

Reply:
it is stored on env side (globally, in db: Environment -> EnvironmentTelemetry -> EnvironmentLogging, from api, it's similar), like s3://cloudbreak-group/logs , for every specific cluster (DH/DL), a path is generated from the environment setting (like s3://cloudbreak-group/logs/datalake/... ) - the specific one is immutable

if the stack is already persisted (for SDX/DistroX) and you can access that, then it's inside the components table with the generated full path (as i mentioned above), for that, see: https://github.com/hortonworks/cloudbreak/blob/master/core/src/main/java/com/sequenceiq/cloudbreak/service/ComponentConfigProviderService.java#L58

Reply:
yea @attilapalfi92 shared some of this detail with me as well. The issue is that its not already persisted. Trying to do this validation before hand (before waiting for RDS to be spun up). I see where the telemetry is being setup. Let me create a separate jira for this. It might not make sense to validate this in the same place.

Reply:
https://jira.cloudera.com/browse/CB-4476